src/casts/convert-closures.rkt

```diff
@@ -1,551 +1,2315 @@
-        
-        [(Labels (app (cc-bnd-code* recur) b*) (app recur e))
-         (Labels b* e)]
-        [(App-Code (app recur e) (app recur* e*))
-         (App-Code e e*)]
-        [(Code-Label u)  (Code-Label u)]
-
-        ;; Coercion Representation Stuff
-        [(Quote-Coercion c)
-         (Quote-Coercion c)]
-        [(HC (app recur p?) (app recur t1) (app recur lbl)
-             (app recur i?) (app recur t2)
-             (app recur m))
-         (HC p? t1 lbl i? t2 m)]
-        [(HC-Inject-Huh (app recur h)) (HC-Inject-Huh h)]
-        [(HC-Project-Huh (app recur h)) (HC-Project-Huh h)]
-        [(HC-Identity-Huh (app recur h)) (HC-Identity-Huh h)]
-        [(HC-Label (app recur h)) (HC-Label h)]
-        [(HC-T1 (app recur h)) (HC-T1 h)]
-        [(HC-T2 (app recur h)) (HC-T2 h)]
-        [(HC-Med (app recur h)) (HC-Med h)]
-        [(Id-Coercion-Huh (app recur e))
-         (Id-Coercion-Huh e)]
-        [(Fn-Coercion-Huh (app recur e))
-         (Fn-Coercion-Huh e)]
-        [(Make-Fn-Coercion u (app recur e1) (app recur e2) (app recur e3))
-         (Make-Fn-Coercion u e1 e2 e3)]
-        [(Fn-Coercion (app recur* e*) (app recur e))
-         (Fn-Coercion e* e)]
-        [(Fn-Coercion-Arity (app recur e))
-         (Fn-Coercion-Arity e)]
-        [(Fn-Coercion-Arg (app recur e1)(app recur e2))
-         (Fn-Coercion-Arg e1 e2)]
-        [(Fn-Coercion-Return (app recur e))
-         (Fn-Coercion-Return e)]
-        [(Id-Fn-Coercion (app recur a)) (Id-Fn-Coercion a)]
-        [(Fn-Coercion-Arg-Set! (app recur f) (app recur i) (app recur a))
-         (Fn-Coercion-Arg-Set! f i a)]
-        [(Fn-Coercion-Return-Set! (app recur f) (app recur r))
-         (Fn-Coercion-Return-Set! f r)]
-        [(Tuple-Coercion-Item-Set! (app recur t) (app recur i) (app recur e))
-         (Tuple-Coercion-Item-Set! t i e)]
-        [(Id-Tuple-Coercion (app recur a))
-         (Id-Tuple-Coercion a)]
-        [(Ref-Coercion (app recur e1) (app recur e2) (app recur flag))
-         (Ref-Coercion e1 e2 flag)]
-        [(Ref-Coercion-Huh (app recur e))
-         (Ref-Coercion-Huh e)]
-        [(Ref-Coercion-Read (app recur e))
-         (Ref-Coercion-Read e)]
-        [(Ref-Coercion-Write (app recur e))
-         (Ref-Coercion-Write e)]
-        [(Ref-Coercion-Ref-Huh (app recur e))
-         (Ref-Coercion-Ref-Huh e)]
-        [(Sequence-Coercion (app recur e1) (app recur e2))
-         (Sequence-Coercion e1 e2)]
-        [(Sequence-Coercion-Huh (app recur e))
-         (Sequence-Coercion-Huh e)]
-        [(Sequence-Coercion-Fst (app recur e))
-         (Sequence-Coercion-Fst e)]
-        [(Sequence-Coercion-Snd (app recur e))
-         (Sequence-Coercion-Snd e)]
-        [(Project-Coercion (app recur e1) (app recur e2))
-         (Project-Coercion e1 e2)]
-        [(Project-Coercion-Huh (app recur e))
-         (Project-Coercion-Huh e)]
-        [(Project-Coercion-Type (app recur e))
-         (Project-Coercion-Type e)]
-        [(Project-Coer
```

Comment:
what are these stack utilities for?

Reply:
They where needed at one point then I abstracted Tarjan's algorithm into a library. I will delete them.